butane_core/src/migrations/mod.rs

```diff
@@ -281,7 +281,7 @@ impl DataObject for ButaneMigration {
 
 impl crate::internal::DataObjectInternal for ButaneMigration {
     const NON_AUTO_COLUMNS: &'static [Column] = Self::COLUMNS;
-    fn pk_mut(&mut self) -> &mut String {
+    fn pk_mut(&mut self) -> &mut impl PrimaryKeyType {
```

Comment:
from nightly
```
warning: impl trait in impl method signature does not match trait method signature
   --> butane_core/src/migrations/mod.rs:284:29
    |
284 |     fn pk_mut(&mut self) -> &mut String {
    |                             ^^^^^^^^^^^
    |
   ::: butane_core/src/lib.rs:70:33
    |
70  |         fn pk_mut(&mut self) -> &mut impl PrimaryKeyType;
    |                                 ------------------------ return type from trait method defined here
    |
    = note: add `#[allow(refining_impl_trait)]` if it is intended for this to be part of the public API of this crate
    = note: we are soliciting feedback, see issue #121718 <https://github.com/rust-lang/rust/issues/121718> for more information
    = note: `#[warn(refining_impl_trait_internal)]` on by default
help: replace the return type so that it matches the trait
    |
284 |     fn pk_mut(&mut self) -> &mut impl sqlval::PrimaryKeyType {
    |                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```